src/main/java/org/osiam/resources/scim/Name.java

```diff
@@ -0,0 +1,327 @@
+     * href="http://tools.ietf.org/html/draft-ietf-scim-core-schema-02#section-6">SCIM core schema 2.0, section 6</a>
+     * </p>
+     *
+     * @return the honorific prefix
+     */
+    public String getHonorificPrefix() {
+        return honorificPrefix;
+    }
+
+    /**
+     * Gets the honorific suffix(es) of the User.
+     * <p>
+     * For more detailed information please look at the <a
+     * href="http://tools.ietf.org/html/draft-ietf-scim-core-schema-02#section-6">SCIM core schema 2.0, section 6</a>
+     * </p>
+     *
+     * @return the honorific sufix
+     */
+    public String getHonorificSuffix() {
+        return honorificSuffix;
+    }
+
+    /**
+     * <p>Checks if this {@link Name} is empty, i.e. all properties are empty or null.</p>
+     *
+     * @return true if all properties are null or empty, else false
+     */
+    @JsonIgnore
+    public boolean isEmpty() {
+        if (!Strings.isNullOrEmpty(formatted)) {
+            return false;
+        }
+
+        if (!Strings.isNullOrEmpty(familyName)) {
+            return false;
+        }
+
+        if (!Strings.isNullOrEmpty(givenName)) {
+            return false;
+        }
+
+        if (!Strings.isNullOrEmpty(middleName)) {
+            return false;
+        }
+
+        if (!Strings.isNullOrEmpty(honorificPrefix)) {
+            return false;
+        }
+
+        if (!Strings.isNullOrEmpty(honorificSuffix)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    @Override
+    public String toString() {
+        return "Name [formatted=" + formatted + ", familyName=" + familyName + ", givenName=" + givenName
+                + ", middleName=" + middleName + ", honorificPrefix=" + honorificPrefix + ", honorificSuffix="
+                + honorificSuffix + "]";
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        Name other = (Name) o;
+
+        if (familyName != null ? !familyName.equals(other.familyName) : other.familyName != null) {
+            return false;
+        }
+        if (formatted != null ? !formatted.equals(other.formatted) : other.formatted != null) {
+            return false;
+        }
+        if (givenName != null ? !givenName.equals(other.givenName) : other.givenName != null) {
+            return false;
+        }
+        if (honorificPrefix != null ? !honorificPrefix.equals(other.honorificPrefix) : other.honorificPrefix != null) {
+            return false;
+        }
+        if (honorificSuffix != null ? !honorificSuffix.equals(other.honorificSuffix) : other.honorificSuffix != null) {
+            return false;
+        }
+        if (middleName != null ? !middleName.equals(other.middleName) : other.middleName != null) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method hashCode() has an NPath complexity of 15625](https://www.codacy.com/app/OSIAM/osiam/file/1979525067/issues/source?bid=3080997&fileBranchId=3117040#l229)